src/modules/AllWidgets.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+
+namespace WPML\PB\Elementor\Modules;
+
+
+class AllWidgets extends \WPML_Elementor_Media_Node {
+
+	/**
+	 * @param array $settings
+	 * @param string $target_lang
+	 * @param string $source_lang
+	 *
+	 * @return array
+	 */
+	public function translate( $settings, $target_lang, $source_lang ) {
+		global $sitepress;
+		$background = [ 'background_image', 'background_overlay_image', 'background_hover_image'] ;
+		foreach ( $background as $element ) {
+			if ( isset( $settings[$element]["url"] ) ) {
+				$media_translate                     = new \WPML_Media_Image_Translate( $sitepress,
```

Comment:
@GeorgeBotsev, you don't need to instantiate this object because you already have it as a dependency:

```php
$settings[$element]["url"] = $this->media_translate->get_translated_image_by_url( $settings[$element]["url"], $source_lang, $target_lang );
```

And actually we already have a better parent protected method for this so you can simplify with:

```php
$backgrounds = [
    'background_image',
    'background_overlay_image',
    'background_hover_image'
] ;

foreach ( $backgrounds as $background ) {
    $settings = $this->media_translate->translate_image_property( $settings, $background, $target_lang, $source_lang );
}

return $settings;
```